paddle/fluid/operators/matmul_op.cc

```diff
@@ -60,7 +60,18 @@ class MatMulKernel : public framework::OpKernel<T> {
     auto mat_dim_b = math::CreateMatrixDescriptor(
         ColumnMatrixFromVector(y.dims()), 0, context.Attr<bool>("transpose_Y"));
     auto scale = static_cast<T>(context.Attr<float>("alpha"));
+
+#if defined(PADDLE_WITH_MKLML) && !defined(PADDLE_WITH_CUDA)
+    int head_number = context.Attr<int>("head_number");
+    if (1 == head_number) {
+      blas.MatMul(x, mat_dim_a, y, mat_dim_b, scale, out, T(0));
+    } else {
+      blas.MatMulWithHead(x, mat_dim_a, y, mat_dim_b, scale, head_number, out,
```

Comment:
Yes. You are right. In normal if head_number > 1, this means we should call MatMulWithHead(). But  in current design, the attribute of  head_number only exist for MKLML and non-CUDA environments. So i have to use macro to limit this judgement.

Reply:
Yes. You are right. In normal if head_number > 1, this means we should call MatMulWithHead(). But in current design, the attribute of head_number only exist for MKLML and non-CUDA environments. So i have to use macro to limit this judgement.

Reply:
Sorry, I meant leaving out the `else` only and keeping the `#else`.

Reply:
Sorry still not quite get your idea on how to remove the else and keep the #else. 
Even using `if (head_number > 1)`, we have to consider the case when `head_number == 1` (this could be the default case when`defined(PADDLE_WITH_MKLML) && !defined(PADDLE_WITH_CUDA)` is true but multiple head is not used. so we still need an else in this case.

Reply:
Right, I mixed it up, sorry.